lib/like-selector.js

```diff
@@ -1,7 +1,8 @@
 export function isLikeSelector(selector) {
+	const prototype = Reflect.getPrototypeOf(selector)
 	return selector !== null
 		&& typeof selector === 'object'
-		&& Reflect.getPrototypeOf(selector) === Object.prototype
+		&& (prototype === Object.prototype || prototype === Array.prototype)
```

Comment:
I think this could use `Array.isArray(selector)`. Note that `Reflect.ownKeys()` on an array returns `length` which messes up the next condition.